src/sphinx_codelinks/sphinx_extension/directives/src_trace.py

```diff
@@ -33,6 +35,42 @@
 logger = logging.getLogger(__name__)
 
 
+def _check_id(
+    config: _SphinxConfig,
+    id: str | None,
+    src_strings: list[str],
+    options: dict[str, str],
+    extra_options: dict[str, str],
+) -> None:
+    """Check and set the id for the need.
+
+    src_strings[0] is always the title.
+    src_strings[1] is always the project.
+    """
+    if config.needs_id_required:
```

Comment:
May this attribute not exist? Asked differently, does the extension make sure `sphinx_need` is registered in `app.extensions`?